src/MSALWrapper/PCACache.cs

```diff
@@ -44,7 +44,16 @@ internal PCACache(ILogger logger, Guid tenantId, string osxKeyChainSuffix = null
             this.osxKeyChainSuffix = string.IsNullOrWhiteSpace(osxKeyChainSuffix) ? $"{tenantId}" : $"{osxKeyChainSuffix}.{tenantId}";
             string appData = Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData);
             this.cacheDir = Path.Combine(appData, ".IdentityService");
-            this.cacheFileName = $"msal_{tenantId}.cache";
+
+            var azureAuthCacheFile = Environment.GetEnvironmentVariable(Constants.AZUREAUTH_CACHE_FILE);
+            if (string.IsNullOrEmpty(azureAuthCacheFile))
+            {
+                this.cacheFileName = $"msal_{tenantId}.cache";
+            }
+            else
+            {
+                this.cacheFileName = azureAuthCacheFile;
+            }

```

Comment:
I don't always love them, but I think this is one case where a ternary would actually improve things.

```suggestion
            this.cacheFileName = string.IsNullOrEmpty(azureAuthCacheFile) ? $"msal_{tenantId}.cache" : azureAuthCacheFile;
```

Reply:
Also, should we be using `string.IsNullOrWhiteSpace()` instead of `string.IsNullOrEmpty()` here? I don't think it would be good for us to allow a filename of `" "`.